src/main/java/net/coderbot/iris/gl/shader/ProgramCreator.java

```diff
@@ -18,6 +18,7 @@ public static int create(String name, GlShader... shaders) {
 
 		// TODO: This is *really* hardcoded, we need to refactor this to support external calls
 		// to glBindAttribLocation
+		GlStateManager._glBindAttribLocation(program, 11, "iris_Entity");
 		GlStateManager._glBindAttribLocation(program, 11, "mc_Entity");
```

Comment:
It don't think `iris_Entity` and `mc_Entity` should both be bound at `11`

Reply:
We need to preserve attributes locations.

Reply:
Wouldn't they conflict if they were both bound at the same location?
or am I missing something here

Reply:
Ah

Reply:
> We need to preserve attributes locations.

Why? Shaders shouldn't be dependent on this information at all. If they are we should definitely be patching them to not be dependent.

Reply:
By preserving locations, I mean we only have 16 of them. We should be reusing when possible.